### DISCUSSION TITLE
Add an interface or abstract class to the TaxDetector

### DISCUSSION THREAD

**Discussion by ghost** [shopware/platform#3447] [RFC / Proposal]
Due to a missing interface or abstract class, the TaxDetector is not extendable, at least multiple times via decorators.
It would be nice if TaxDetector could be decorated correctly since that was kept in mind at many other places.

**Comment by OliverSkroblin** [comment]
Hey @ezycod,

I added an abstract class in 6.6: https://github.com/shopware/shopware/commit/5d060e2b8febc403b8e416c4998cc01a95c273df

Thanks for the RFC 👋

**Reply by OliverSkroblin** [reply]
Backport in 6.5: https://github.com/shopware/shopware/commit/ab937583d0dea481cb916b343c81db1bc021f40d
